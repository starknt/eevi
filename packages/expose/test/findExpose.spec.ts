import { expect, test } from 'vitest'
import { findExports } from 'mlly'
import MagicString from 'magic-string'

test('find expose', () => {
  const code = `
    export const name = 'test'
    export function testfnc() {
      console.log('hello')
    }
    const a = 'hello'
    const b = 'world'
    export { a, b }

    const d = 123
    export default d

    export * from './file'
  `

  const transformed = new MagicString(code)
  const exports = findExports(code)

  for (const ex of exports) {
    switch (ex.type) {
      case 'declaration':
        break
      case 'default':
        break
      case 'named':
        break
      case 'star':
        break
    }
  }

  expect(findExports(code)).toMatchSnapshot()
})
