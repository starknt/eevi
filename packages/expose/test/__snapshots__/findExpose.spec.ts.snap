// Vitest Snapshot v1

exports[`find expose > 
    [
      {
        "code": "export const name",
        "declaration": "const",
        "end": 22,
        "name": "name",
        "names": [
          "name",
        ],
        "start": 5,
        "type": "declaration",
      },
      {
        "code": "export function testfnc",
        "declaration": "function",
        "end": 59,
        "name": "testfnc",
        "names": [
          "testfnc",
        ],
        "start": 36,
        "type": "declaration",
      },
      {
        "code": "export { a, b }",
        "end": 160,
        "exports": " a, b",
        "names": [
          "a",
          "b",
        ],
        "specifier": undefined,
        "start": 145,
        "type": "named",
      },
    ]
   1`] = `
[
  {
    "code": "export const name",
    "declaration": "const",
    "end": 22,
    "name": "name",
    "names": [
      "name",
    ],
    "start": 5,
    "type": "declaration",
  },
  {
    "code": "export function testfnc",
    "declaration": "function",
    "end": 59,
    "name": "testfnc",
    "names": [
      "testfnc",
    ],
    "start": 36,
    "type": "declaration",
  },
  {
    "code": "export { a, b }",
    "end": 160,
    "exports": " a, b",
    "names": [
      "a",
      "b",
    ],
    "specifier": undefined,
    "start": 145,
    "type": "named",
  },
  {
    "code": "export default ",
    "end": 199,
    "name": "default",
    "names": [
      "default",
    ],
    "start": 184,
    "type": "default",
  },
  {
    "code": "export * from './file'",
    "end": 228,
    "name": undefined,
    "specifier": "./file",
    "start": 206,
    "type": "star",
  },
]
`;
